eva_submission/eload_deletion.py

```diff
@@ -0,0 +1,98 @@
+import os
+import shutil
+import tarfile
+from pathlib import Path
+
+from ebi_eva_common_pyutils.config import cfg
+
+from eva_submission.eload_submission import Eload
+from eva_submission.submission_in_ftp import deposit_box
+
+
+class EloadDeletion(Eload):
+    def __init__(self, eload_number):
+        super().__init__(eload_number)
+        self.project_accession = self.eload_cfg.query('brokering', 'ena', 'PROJECT')
+        self.project_dir = os.path.join(cfg['projects_dir'], self.project_accession)
+
+    def delete_submission(self, ftp_box, submitter):
+        self.upgrade_to_new_version_if_needed()
+        self.archive_eload()
+
+        # delete
+        ftp_dir = deposit_box(ftp_box, submitter)
+        self.delete_ftp_dir(ftp_dir)
+        self.delete_project_dir(self.project_dir)
+        self.delete_eload_dir(self.eload_dir)
+
+    def is_compressed(self, file_name):
+        compressed_exts = (".gz", ".xz", ".bz2", ".zip", ".rar", ".7z")
+        return file_name.endswith(compressed_exts)
+
+    def archive_eload(self):
+        archive_dir = os.path.join(self.eload_dir, 'archive_dir')
+        os.makedirs(archive_dir, exist_ok=True)
+        archive_file = os.path.join(self.eload_dir, f'{self.eload}.tar')
+        # copy relevant files to the archive_dir
+        self.copy_eload_files(archive_dir)
+
+        # archive eload
+        with tarfile.open(archive_file, "w:gz") as tar:
+            for root, _, files in os.walk(archive_dir):
+                for file in files:
+                    file_path = os.path.join(root, file)
+                    arcname = os.path.relpath(file_path, start=archive_dir)
+                    if self.is_compressed(file):
+                        tar.add(file_path, arcname=arcname, recursive=False, filter=lambda x: x)
+                    else:
+                        tar.add(file_path, arcname=arcname)
+
+        # copy to lts
+        shutil.copy(archive_file, cfg['eloads_lts_dir'])
+
+    def copy_eload_files(self, archive_dir):
```

Comment:
When we manually archived these files previously, we compress the files and didn't apply any compression in the tar file generation.
That might be an option here as well.